usaspending_api/references/v2/views/recipients.py

```diff
@@ -0,0 +1,110 @@
+from collections import OrderedDict
+
+from elasticsearch_dsl import Q as ES_Q
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import RecipientSearch
+from usaspending_api.common.validator.tinyshield import validate_post_request
+from usaspending_api.search.v2.es_sanitization import es_sanitize
+
+models = [
+    {"name": "search_text", "key": "search_text", "type": "text", "text_type": "search", "optional": False},
+    {"name": "limit", "key": "limit", "type": "integer", "max": 500, "optional": True, "default": 10},
+    {
+        "name": "recipient_levels",
+        "key": "recipient_levels",
+        "type": "array",
+        "array_type": "text",
+        "text_type": "search",
+        "items": {"type": "string"},
+        "optional": True,
+    },
+]
+
+
+@validate_post_request(models)
+class RecipientAutocompleteViewSet(APIView):
+    """
+    This endpoint is used for the Recipient autocomplete filter which returns a list of recipients matching the
+    specified search text.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/autocomplete/recipient.md"
+
+    @cache_response()
+    def post(self, request, format=None):
+        search_text, recipient_levels = prepare_search_terms(request.data)
+        limit = request.data["limit"]
+        query = create_es_search(search_text, recipient_levels, limit)
+        results = query_elasticsearch(query)
+        response = OrderedDict([("count", len(results)), ("results", results)])
+        return Response(response)
+
+
+def prepare_search_terms(request_data):
+    fields = [request_data["search_text"], request_data.get("recipient_levels", [])]
+    return [es_sanitize(field).upper() if isinstance(field, str) else field for field in fields]
+
+
+def create_es_search(search_text, recipient_levels, limit):
+    ES_RECIPIENT_SEARCH_FIELDS = ["recipient_name", "uei"]
+
+    query = ES_Q()
+    # Build the should_clauses list
+    search_text_should_clause = [
+        ES_Q(
+            "bool",
+            should=[
+                ES_Q("query_string", query=search_text, fields=ES_RECIPIENT_SEARCH_FIELDS),
+                ES_Q("match", recipient_name=search_text),
+                ES_Q("match", uei=search_text),
+            ],
+            minimum_should_match=1,
+        )
+    ]
+
+    if recipient_levels:
+        recipient_should_clause = [
+            ES_Q(
+                "bool",
+                should=[ES_Q("match", recipient_level=level) for level in recipient_levels],
+                minimum_should_match=1,
+            )
+        ]
+
+        query = ES_Q(
+            "bool", must=[ES_Q("bool", should=recipient_should_clause), ES_Q("bool", should=search_text_should_clause)]
+        )
+
+    else:
```

Comment:
Instead of an `if...else` here. Why not just initialize the variable `query` to `search_text_should_clause` and include a single `if` statement. If the `if` statement evaluates to true then it will override the initial value. Doing so will make code that's easier to read (especially since the else is one line).

Reply:
If the if statement evaluates to true, we don't want to override the initial value. if there are recipient_levels, the ES_Q query has a must with two shoulds inside (search_text_should_clause) & (recipient_should_clause). the else portion does not have to worry about the recipient_should_clause. Need the if-else

Reply:
I guess I could set it to override when the if statement evaluates to true, I would just be reusing the query and setting the should=query (on line 78 where we need to set the should =  search_text_should_clause). But this would be confusing for future developers.

Reply:
Hmm I don't think you understand my point. Leave the `if` remove the `else` and move the contents from the `else` right above the `if`.

Reply:
OH thats what you meant. Okay will do!